client/src/partials/fiscal/fiscal.openingBalance.js

```diff
@@ -0,0 +1,173 @@
+
+  // expose to the view
+  vm.editBalanceEnabled  = false;
+  vm.showAccountFilter   = false;
+  vm.toggleEditBalance   = toggleEditBalance;
+  vm.toggleAccountFilter = toggleAccountFilter;
+  vm.submit = submit;
+
+  // grid options
+  vm.indentTitleSpace = 20;
+  vm.gridApi = {};
+
+  var columns = [
+    { field : 'number', displayName : '', cellClass : 'text-right', width : 100},
+    { field : 'label',
+      displayName : 'FORM.LABELS.ACCOUNT',
+      cellTemplate : '/partials/accounts/templates/grid.labelCell.tmpl.html',
+      headerCellFilter : 'translate',
+      enableFiltering: true,
+      enableColumnMenu: false
+    },
+    { field : 'debit',
+      displayName : 'FORM.LABELS.DEBIT',
+      headerCellClass : 'text-center',
+      headerCellFilter : 'translate',
+      cellTemplate : '/partials/fiscal/templates/balance.debit.tmpl.html',
+      width : 200,
+      enableFiltering: false,
+      enableColumnMenu: false
+    },
+    { field : 'credit',
+      displayName : 'FORM.LABELS.CREDIT',
+      headerCellClass : 'text-center',
+      headerCellFilter : 'translate',
+      cellTemplate : '/partials/fiscal/templates/balance.credit.tmpl.html',
+      width : 200,
+      enableFiltering: false,
+      enableColumnMenu: false
+    }
+  ];
+
+  vm.gridOptions = {
+    appScopeProvider : vm,
+    enableSorting : false,
+    enableFiltering : vm.showAccountFilter,
+    showTreeExpandNoChildren : false,
+    enableColumnMenus : false,
+    rowTemplate : '/partials/accounts/templates/grid.titleRow.tmpl.html',
+    columnDefs : columns,
+    onRegisterApi : onRegisterApi
+  };
+
+  // API register function
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // get fiscal year
+  Fiscal.read(fiscalYearId)
+  .then(function (fy) {
+    vm.fiscal = fy;
+    return fy.previous_fiscal_year_id;
+  })
+  .then(hasPrevious)
+  .catch(Notify.handleError);
+
+  // get account
+  Fiscal.periodicBalance({
+    id: fiscalYearId,
+    period_number: 0
+  })
+  .then(function (list) {
+    vm.accounts = list;
+    vm.gridOptions.data = Accounts.order(vm.accounts);
+  })
+  .catch(Notify.handleError);
+
+  /**
+   * @function submit
+   */
+  function submit(form) {
+    vm.balanced = hasBalancedAccount();
+
+    if (!vm.balanced) { return ; }
+
+    // set the fiscal year opening balance
+    Fiscal.setOpeningBalance({
+      id: fiscalYearId,
+      fiscal: vm.fiscal,
+      accounts: vm.accounts
+    })
+    .then(function () {
+      Notify.success(vm.previousFiscalYearExist ? 'FORM.INFO.IMPORT_SUCCESS' : 'FORM.INFO.SAVE_SUCCESS');
+    })
+    .catch(Notify.handleError);
+
+  }
+
+  /**
+   * @function toggleAccountFilter
+   * @description show a filter for finding an account
+   */
+  function toggleAccountFilter() {
+    vm.showAccountFilter = !vm.showAccountFilter;
+    vm.gridOptions.enableFiltering = vm.showAccountFilter;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.ALL);
```

Comment:
It would be more efficient to simply signal that the options have changed with `uiGridConstants.dataChange.OPTIONS`.  See if that works.

[Relevant Documentation](http://ui-grid.info/docs/#/api/ui.grid.service:uiGridConstants#properties_datachange)